vscode/vscode/vscode.ml

```diff
@@ -2521,6 +2652,271 @@ module TreeView = struct
+
+    val postMessage : t -> Js.Any.t -> bool Promise.t [@@js.call]
+
+    val create :
+         onDidReceiveMessage:OnDidReceiveMessage.t
+      -> cspSource:string
+      -> html:string
+      -> options:WebviewOptions.t
+      -> close:(unit -> unit)
+      -> asWebviewUri:(Uri.t -> Uri.t)
+      -> postMessage:(Js.Any.t -> bool Promise.t)
+      -> t
+      [@@js.builder]]
+end
+
+module rec WebviewPanel : sig
+  include Js.T
+
+  module LightDarkIcon : sig
+    type t =
+      { light : [ `Uri of Uri.t ]
+      ; dark : [ `Uri of Uri.t ]
+      }
+
+    include Js.T with type t := t
+  end
+
+  val onDidChangeViewState :
+    t -> WebviewPanelOnDidChangeViewStateEvent.t Event.t
+
+  val onDidDispose : t -> unit Event.t
+
+  val active : t -> bool
+
+  type iconPath =
+    [ `Uri of Uri.t
+    | `LightDark of LightDarkIcon.t
+    ]
+
+  val options : t -> WebviewPanelOptions.t
+
+  val title : t -> string
+
+  val viewColumn : t -> ViewColumn.t
+
+  val viewType : t -> string
+
+  val visible : t -> bool
+
+  val webview : t -> WebView.t
+
+  val set_webview : t -> WebView.t -> unit
+
+  val dispose : t -> Js.Any.t
+
+  val reveal :
+    t -> ?preserveFocus:bool -> ?viewColumn:ViewColumn.t -> unit -> unit
+
+  val create :
+       onDidChangeViewState:WebviewPanelOnDidChangeViewStateEvent.t Event.t
+    -> onDidDispose:unit Event.t
+    -> active:bool
+    -> options:WebviewPanelOptions.t
+    -> title:string
+    -> viewColumn:ViewColumn.t
+    -> viewType:string
+    -> visible:bool
+    -> webview:WebView.t
+    -> dispose:Js.Any.t
+    -> reveal:(?preserveFocus:bool -> ?viewColumn:ViewColumn.t -> unit -> unit)
+    -> t
+end = struct
+  include Interface.Make ()
+
+  module LightDarkIcon = struct
+    type t =
+      { light : ([ `Uri of Uri.t ][@js.union])
+      ; dark : ([ `Uri of Uri.t ][@js.union])
+      }
+    [@@js]
+
+    let t_of_js js_val =
+      let light_js = Ojs.get_prop_ascii js_val "light" in
+      let dark_js = Ojs.get_prop_ascii js_val "dark" in
+      let light = `Uri ([%js.to: Uri.t] light_js) in
+      let dark = `Uri ([%js.to: Uri.t] dark_js) in
+      { light; dark }
+  end
+
+  type iconPath =
+    ([ `Uri of Uri.t
+     | `LightDark of LightDarkIcon.t
+     ]
+    [@js.union])
+  [@@js]
+
+  module OnDidChangeViewState =
+    Event.Make (WebviewPanelOnDidChangeViewStateEvent)
+  module OnDidDispose = Event.Make (Js.Unit)
+
+  include
+    [%js:
+    val onDidChangeViewState : t -> OnDidChangeViewState.t [@@js.get]
+
+    val onDidDispose : t -> OnDidDispose.t [@@js.get]
+
+    (*let iconPath_of_js js_val = if Ojs.has_property js_val "path" then `Uri
+      ([%js.to: Uri.t] js_val) else if Ojs.has_property js_val "light" then
+      `LightDark ([%js.to: LightDarkIcon.t] js_val) else assert false*)
+
+    val active : t -> bool [@@js.get]
+
+    val options : t -> WebviewPanelOptions.t [@@js.get]
+
+    val title : t -> string [@@js.get]
+
+    val viewColumn : t -> ViewColumn.t [@@js.get]
```

Comment:
Possibly undefined. Should return `ViewColumn.t or_undefined`.

https://github.com/microsoft/vscode/blob/1.58.2/src/vs/vscode.d.ts#L7583

Reply:
Done in b0f217bbb0a6f8a866299ee40e4122dee1f2a1dd